packages/vite/src/node/server/transformRequest.ts

```diff
@@ -49,7 +49,7 @@ export async function transformRequest(
   const module = await moduleGraph.getModuleByUrl(url)
   const cached =
     module && (ssr ? module.ssrTransformResult : module.transformResult)
-  if (cached) {
+  if (cached && !options.html) {
```

Comment:
I don't think it is a better fix. Eg, you have a  html plugins which transform html files and you visit it twice, this plugins will run twice. It is caused by if you visit `localhost:3000/main` and expect html fallback, but vite will try match some extensions like `.js/ts`. You should add `html` flag for it.

Reply:
@underfin Thanks for the advice，I submitted this change because I read the following code vite and misunderstood that vite does not process html files.

```
  const loadResult = await pluginContainer.load(id, ssr)
  if (loadResult == null) {
    // if this is an html request and there is no load result, skip ahead to
    // SPA fallback.
    if (options.html && !id.endsWith('.html')) {
      return null
    }
```
I redefined the problem and found that the vite config `resolve.extensions` allow vite to match more types of files such as · 
 mjs | ts | js | jsx | tsx | json·，but lost the ability to match the original path at the same time.

to fix the issue #3502 , when we want to visit the original path such as ·localhost:3000/route· and treat `/route` as a route instead of a module `route.ts`，we can add any suffix that is not recognized by vite，e.g. `localhost:3000/route?xxxxxxx` or `localhost:3000/route?kkkkkkk` and it will run as expected. But this does not seem to be an official and perfect solution

Reply:
By the way, I am trying to add a html flag as the official standard suffix.